jans-core/document-store/src/main/java/io/jans/service/document/store/provider/DBDocumentStoreProvider.java

```diff
@@ -0,0 +1,213 @@
+	private DBDocumentService documentService;
+	
+	@Inject
+	private DocumentStoreConfiguration documentStoreConfiguration;
+
+	private DBDocumentStoreConfiguration dbDocumentStoreConfiguration;
+
+	public DocumentStoreType getProviderType() {
+		return DocumentStoreType.DB;
+	}
+	
+	@PostConstruct
+	public void init() {
+		this.dbDocumentStoreConfiguration = documentStoreConfiguration.getDbConfiguration();
+	}
+	
+	public void configure(DocumentStoreConfiguration documentStoreConfiguration, PersistenceEntryManager persistenceEntryManager) {
+		this.log = LoggerFactory.getLogger(DBDocumentStoreProvider.class);
+		this.documentStoreConfiguration = documentStoreConfiguration;
+		if(documentService == null) {
+			this.documentService = new DBDocumentService(persistenceEntryManager);
+		}
+	}
+
+	@Override
+	public void create() {
+		
+	}
+
+	@Override
+	public void destroy() {
+		
+	}
+
+	@Override
+	public boolean hasDocument(String DisplayName) {
+		log.debug("Has document: '{}'", DisplayName);
+		if (StringHelper.isEmpty(DisplayName)) {
+			throw new IllegalArgumentException("Specified path should not be empty!");
+		}		
+		OxDocument oxDocument = null;
+		try {
+			oxDocument = documentService.getOxDocumentByDisplayName(DisplayName);
+		} catch (Exception e) {
+			log.error("Failed to check if path '" + DisplayName + "' exists in repository", e);
+		}
+
+		return false;
+	}
+
+	@Override
+	public boolean saveDocument(String name, String documentContent, Charset charset, List<String> moduleList) {
+		log.debug("Save document: '{}'", name);
+		OxDocument oxDocument = new OxDocument();
+		oxDocument.setDocument(documentContent);
+		oxDocument.setDisplayName(name);		
+		try {
+			try {
+				oxDocument.setInum(documentService.generateInumForNewOxDocument());	
+				String dn = "inum="+ oxDocument.getInum() +",ou=document,o=gluu";
+				oxDocument.setDn(dn);
+				oxDocument.setDescription(name);
+				oxDocument.setOxEnabled("true");
+				oxDocument.setOxModuleProperty(moduleList);	  
+				documentService.addOxDocument(oxDocument);
+				return true;
+			} finally {
+			}
+		} catch (Exception ex) {
+			log.error("Failed to write document to file '{}'", name, ex);
+		}
+
+		return false;
+	}
+
+	@Override
+	public boolean saveDocumentStream(String name, InputStream documentStream, List <String> moduleList) {
+		
+		OxDocument oxDocument = new OxDocument();
+		oxDocument.setDisplayName(name);
+		
+		 try {
+			String documentContent = Base64.getEncoder().encodeToString(IOUtils.toByteArray(documentStream));
+			oxDocument.setDocument(documentContent);
+			String inum = documentService.generateInumForNewOxDocument();
+			oxDocument.setInum(inum);	
+			String dn = "inum="+ oxDocument.getInum() +",ou=document,o=gluu";
+			oxDocument.setDn(dn);
+			oxDocument.setDescription(name);
+			oxDocument.setOxEnabled("true");
+			oxDocument.setOxModuleProperty(moduleList);
+			documentService.addOxDocument(oxDocument);
+			return true;
+		} catch (IOException e) {
```

Comment:
IOException is redundant because it has exactly same statements as super `Exception`.